examples/multi_invokee_rendering/source/multi_invokee_rendering.cpp

```diff
@@ -0,0 +1,200 @@
+		: gvk::invokee(pExecutionOrder), mQueue{ &aQueue }, trianglePart{ trianglePart }, invokeeList{ invokeeList }
+	{}
+
+	void initialize() override
+	{
+		// the invokee which receives the other invokees list to create the UI
+		bool isGUIBuilder = invokeeList.has_value();
+		
+		if (isGUIBuilder)
+		{
+			invokeeList->push_back(this);
+			// Print some information about the available memory on the selected physical device:
+			gvk::context().print_available_memory_types();
+		}
+		auto wnd = gvk::context().main_window();
+
+		std::vector<avk::attachment> attachments;
+		attachments.push_back(avk::attachment::declare(gvk::format_from_window_color_buffer(gvk::context().main_window()), avk::on_load::load, avk::color(0), avk::on_store::store)); // But not in presentable format, because ImGui (and other invokees) comes after);
+		for (auto a : wnd->get_additional_back_buffer_attachments()) {
+			a.mLoadOperation = avk::on_load::dont_care;
+			a.mStoreOperation = avk::on_store::dont_care;
+			attachments.push_back(a);
+		}
+		mRenderPass = gvk::context().create_renderpass(
+			attachments,
+			[](avk::renderpass_sync& rpSync) {
+				if (rpSync.is_external_pre_sync()) {
+					rpSync.mSourceStage = avk::pipeline_stage::color_attachment_output;
+					rpSync.mSourceMemoryDependency = avk::memory_access::color_attachment_write_access;
+					rpSync.mDestinationStage = avk::pipeline_stage::color_attachment_output;
+					rpSync.mDestinationMemoryDependency = avk::memory_access::color_attachment_read_access;
+				}
+				if (rpSync.is_external_post_sync()) {
+					rpSync.mSourceStage = avk::pipeline_stage::color_attachment_output;
+					rpSync.mSourceMemoryDependency = avk::memory_access::color_attachment_write_access;
+					rpSync.mDestinationStage = avk::pipeline_stage::bottom_of_pipe;
+					rpSync.mDestinationMemoryDependency = {};
+				}
+			}
+		);
+
+
+		// Create a graphics pipeline:
+		mPipeline = gvk::context().create_graphics_pipeline_for(
+			avk::vertex_shader("shaders/a_triangle.vert"),
+			avk::fragment_shader("shaders/a_triangle.frag"),
+			avk::cfg::front_face::define_front_faces_to_be_clockwise(),
+			avk::cfg::viewport_depth_scissors_config::from_framebuffer(gvk::context().main_window()->backbuffer_at_index(0)),
+			avk::attachment::declare(gvk::format_from_window_color_buffer(gvk::context().main_window()), avk::on_load::load, avk::color(0), avk::on_store::store),
+			avk::push_constant_binding_data{ avk::shader_type::vertex, 0, sizeof(unsigned int) }
+		);
+
+		// set up updater
+		// we want to use an updater, so create one:
+		mUpdater.emplace();
+		mPipeline.enable_shared_ownership(); // Make it usable with the updater
+		mUpdater->on(
+				gvk::swapchain_resized_event(gvk::context().main_window()),
+				gvk::shader_files_changed_event(mPipeline)
+			)
+			.update(mPipeline);				
+
+		if (isGUIBuilder)
```

Comment:
Hmm, this whole `isGUIBuilder` (which would optimally be named `isGuiBuilder` b.t.w., but doesn't matter since we can hopefully get rid of it)  strikes me as overly complicated. Also, the passed `invokeeList`  is pretty confusing and there must be a better way for a "basic example".

Here is a suggestion that should (hopefully) work:
Multiple callbacks can be added to an `imgui_manager`, i.e. that would be one `imguiManager->add_callback` per `invokee`. 
The callbacks are executed in the order of their addition.
If everything shall be drawn in one ImGui window, then one could add another `imguiManager->add_callback` directly in `main()` which invokes only `ImGui::Begin` (but I don't really know how the `ImGui::End` could be added elegantly).

Reply:
I wanted to talk to you extensively about this one actually.

The issues in my opinion is originated in the way that ImGui currently works with other invokees (dependencies, see below).
I would rather design a more robust UI setup procedure that does not need the init call of the invokees.

This is by itself is the result of the blocking "start" call (which is needed) which then means there is no opportunity to call the UI's setup procedure. I suggest a special callback handler for the UI, or a way to inject code inside UI's invokee Initialize method.

Regarding your above suggestion, I will check that. But I think we cannot add the callbacks before "start" is called (I believe I have tried that already).

Reply:
resolved as discussed previously